src/core/Tree/Tree.tsx

```diff
@@ -0,0 +1,214 @@
+    const nodes: Array<HTMLElement> = Array.from(
+      treeRef.current?.querySelectorAll('.iui-tree-node:not(.iui-disabled)') ||
+        [],
+    );
+
+    if (!nodes.length) {
+      return;
+    }
+
+    let newIndex = -1;
+    const currentIndex = nodes.findIndex(
+      (node) => node === treeRef.current?.ownerDocument.activeElement,
+    );
+
+    const expander = nodes[currentIndex].querySelector(
+      '.iui-tree-node-content-expander-icon',
+    ) as HTMLElement;
+
+    switch (event.key) {
+      case 'ArrowDown': {
+        newIndex = currentIndex + 1;
+        break;
+      }
+      case 'ArrowUp': {
+        newIndex = currentIndex - 1;
+        break;
+      }
+      case 'ArrowLeft':
+        if (
+          expander === null ||
+          !expander?.classList.contains(
+            'iui-tree-node-content-expander-icon-expanded',
+          )
+        ) {
+          let parentIndex = -1;
+          for (let n = 0; n < nodes.length; n++) {
+            const parentOwns = (nodes[n].parentElement
+              ?.lastChild as HTMLElement)
+              .getAttribute('aria-owns')
+              ?.split(', ');
+            if (
+              parentOwns &&
+              parentOwns?.findIndex(
+                (id) => id === nodes[currentIndex].parentElement?.id,
+              ) != -1
+            ) {
+              parentIndex = n;
+              break;
+            }
+          }
+          if (parentIndex != -1) {
+            newIndex = parentIndex;
+          } else {
+            newIndex = currentIndex - 1;
+          }
+        } else {
+          expander.parentElement?.click();
+        }
+        break;
+      case 'ArrowRight':
+        if (
+          expander === null ||
+          expander?.classList.contains(
+            'iui-tree-node-content-expander-icon-expanded',
+          )
+        ) {
+          newIndex = currentIndex + 1;
+        } else {
+          expander.parentElement?.click();
+          return;
+        }
+        break;
+      case 'Enter':
+      case ' ':
+      case 'Spacebar':
+        nodes[currentIndex].click();
+        event.preventDefault();
+        return;
+    }
+
+    if (newIndex >= 0 && newIndex < nodes.length) {
+      nodes[newIndex].focus();
+      event.preventDefault();
+    }
+  };
+
+  const flatNodesList = React.useMemo(() => {
+    const flatList: NodeData<T>[] = [];
+
+    const flatNodes = (nodes: T[] = [], depth = 0) => {
+      const nodeIdList = Array<string>();
+      nodes.forEach((element) => {
+        const flatNode = getNode(element);
+        nodeIdList.push(flatNode.nodeId);
+        flatList.push(flatNode);
+        flatNode.depth = depth;
+        if (flatNode.isExpanded) {
+          const subNodeIds = flatNodes(flatNode.subNodes, depth + 1);
+          flatNode.subNodeIds = subNodeIds;
+        } else {
+          const subNodeIds = Array<string>();
+          flatNode.subNodes?.forEach((subNode) => {
+            subNodeIds.push(subNode.id);
```

Comment:
Build is failing because here  Property 'id' does not exist on type 'T'. These subnode ids are needed for the `aria-owns` label on each parent node. Not sure if I should just call `getNode` on all of the subnodes to get `nodeId` (even if they should be collapsed)

Reply:
Aren't you getting subNodeIds from this part?
![image](https://user-images.githubusercontent.com/36186912/150542544-377cd3fe-6e11-457c-8ceb-20536f41a48b.png)

Reply:
Yes only for expanded parent nodes, the else covers collapsed nodes